tests/posix/sem/src/sem.c

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <semaphore.h>
+#include <kernel.h>
+#include <pthread.h>
+#include <ztest.h>
+
+#define STACK_SIZE 1024
+struct k_thread thread;
+PTHREAD_SEM_DEFINE(sema, 0, 1);
+
+static K_THREAD_STACK_DEFINE(thread_stack, STACK_SIZE);
+
+static void foo_func(void)
+{
+	sem_wait(&sema);
+	printk("Print me after taking semaphore!\n");
+	sem_post(&sema);
+}
+
+static void test_sema(void)
```

Comment:
what exactly are you testing here? You are just running a thread using ztest and basically testing NOTHING, under which circumstances is this test going to fail? where are the asserts?

Reply:
the posix test needs to go under tests/kerne/posix